paasta_tools/deployd/common.py

```diff
@@ -50,7 +80,28 @@ def put(self, item, *args, **kwargs):
         Queue.put(self, item, *args, **kwargs)
 
 
-def rate_limit_instances(instances, number_per_minute, watcher_name):
+class PaastaPriorityQueue(PriorityQueue):
+
+    def __init__(self, name, *args, **kwargs):
+        self.name = name
+        PriorityQueue.__init__(self, *args, **kwargs)
+        self.counter = 0
+
+    @property
+    def log(self):
+        name = '.'.join([self.__class__.__module__, self.__class__.__name__])
```

Comment:
generally prefer `type(self)` over `self.__class__` (python3 actually proposed removing `__class__` as it was leftover from python2 old style classes)